### PR TITLE
Feature/parser code refactoring

### PR DESCRIPTION
**Pull request by kogonidze**

### CODE REVIEW

File: Octopus.Core.Common/Extensions/FileInfoExtensions.cs [hunk 1]
```diff
@@ -24,5 +25,25 @@ public static FileExtension GetFileExtension(this FileInfo file)
                     return FileExtension.Unknown;
             }
         }
+
+        public static bool IsAvailable(this FileInfo file)
+        {
+            try
+            {
+                using (var fs = file.Open(FileMode.Open, FileAccess.Read, FileShare.Read))
```
[the remark this answers is not in the document]
Comment (kogonidze) [reply Octopus.Core.Common/Extensions/FileInfoExtensions.cs:33]: Do you mean such syntax?

`using var fs = file.Open(FileMode.Open, FileAccess.Read, FileShare.Read);`

[the remark this answers is not in the document]
Comment (Anton-Sam) [reply Octopus.Core.Common/Extensions/FileInfoExtensions.cs:33]: Yes)

**Review by Anton-Sam — APPROVED** [review]
Good job!

File: Octopus.Core.Common/Extensions/FileInfoExtensions.cs [hunk 2]
```diff
@@ -24,5 +25,23 @@ public static FileExtension GetFileExtension(this FileInfo file)
                     return FileExtension.Unknown;
             }
         }
+
+        public static bool IsAvailable(this FileInfo file)
+        {
+            try
+            {
+                using var fs = file.Open(FileMode.Open, FileAccess.Read, FileShare.Read);
+            }
```
Comment (VladiTitov) [inline Octopus.Core.Common/Extensions/FileInfoExtensions.cs:34]: Maybe `return true;` can be posted here?

File: Octopus.Core.Common/Extensions/FileInfoExtensions.cs [hunk 3]
```diff
@@ -24,5 +25,23 @@ public static FileExtension GetFileExtension(this FileInfo file)
...
+            catch (IOException)
+            {
+                return false;
+            }
+            catch (Exception)
+            {
+                throw;
+            }
+
+            return true;
```
Comment (VladiTitov) [inline Octopus.Core.Common/Extensions/FileInfoExtensions.cs:44]: and this delete?

File: Octopus.Core.Parser.BusinessLogic/Interfaces/Services/IParserProcessor.cs [hunk 4]
```diff
@@ -1,12 +1,11 @@
...
 
-namespace Octopus.Core.Parser.WorkerService.Interfaces.Services
+namespace Octopus.Core.Parser.BusinessLogic.Interfaces.Services
 {
     public interface IParserProcessor
     {
-        Task Parse(IEntityDescription request);
+        Task ProcessInputData(ParserInputData inputData);
 
         Task StartProcessing();
```
Comment (VladiTitov) [inline Octopus.Core.Parser.BusinessLogic/Interfaces/Services/IParserProcessor.cs:10]: maybe the lines should be swapped for greater readability?

File: Octopus.Core.Parser.BusinessLogic/Services/MessageHandler.cs [hunk 5]
```diff
@@ -11,20 +11,23 @@ public class MessageHandler : IEventProcessor
...
         }
 
         public async void ProcessEvent(string message)
         {
             try
             {
                 _entityDescription = JsonSerializer.Deserialize<EntityDescription>(message);
-                await _parserProcessor.Parse(_entityDescription);
+
+                var parserInputData = await _validationService.ValidateEntityDescription(_entityDescription);
```
Comment (VladiTitov) [inline Octopus.Core.Parser.BusinessLogic/Services/MessageHandler.cs:28]: good idea!

File: Octopus.Core.Parser.BusinessLogic/Services/MessageHandler.cs [hunk 6]
```diff
@@ -11,20 +11,23 @@ public class MessageHandler : IEventProcessor
...
-
-            _parserProcessor.StartProcessing();
+            _validationService = validationService;
         }
 
         public async void ProcessEvent(string message)
         {
             try
             {
                 _entityDescription = JsonSerializer.Deserialize<EntityDescription>(message);
```
Comment (VladiTitov) [inline Octopus.Core.Parser.BusinessLogic/Services/MessageHandler.cs:26]: We have special string extention for this. Use this extension

File: Octopus.Core.Common/Extensions/FileInfoExtensions.cs (hunk 2 above)
Comment (VladiTitov) [inline Octopus.Core.Common/Extensions/FileInfoExtensions.cs:34]: Maybe `return true;` can be posted here?

File: Octopus.Core.Common/Extensions/FileInfoExtensions.cs (hunk 3 above)
Comment (VladiTitov) [inline Octopus.Core.Common/Extensions/FileInfoExtensions.cs:44]: and this delete?

File: Octopus.Core.Parser.BusinessLogic/Interfaces/Services/IParserProcessor.cs (hunk 4 above)
Comment (VladiTitov) [inline Octopus.Core.Parser.BusinessLogic/Interfaces/Services/IParserProcessor.cs:10]: maybe the lines should be swapped for greater readability?

File: Octopus.Core.Parser.BusinessLogic/Services/MessageHandler.cs (hunk 6 above)
Comment (VladiTitov) [inline Octopus.Core.Parser.BusinessLogic/Services/MessageHandler.cs:26]: We have special string extention for this. Use this extension

**Review by VladiTitov — CHANGES_REQUESTED** [review]

File: Octopus.Core.Parser.BusinessLogic/Interfaces/Services/IParserProcessor.cs (hunk 4 above)
Comment (kogonidze) [reply Octopus.Core.Parser.BusinessLogic/Interfaces/Services/IParserProcessor.cs:10]: fixed

File: Octopus.Core.Parser.BusinessLogic/Services/MessageHandler.cs (hunk 5 above)
Comment (kogonidze) [reply Octopus.Core.Parser.BusinessLogic/Services/MessageHandler.cs:28]: thanks!

File: Octopus.Core.Common/Extensions/FileInfoExtensions.cs (hunk 3 above)
Comment (kogonidze) [reply Octopus.Core.Common/Extensions/FileInfoExtensions.cs:44]: fixed

**Review by VladiTitov — APPROVED** [review]
Good job!

**Pull request merged**